### PR TITLE
Change /src/index.js to .jsx and remove Strict Mode

### LINKED ISSUE #287: [Bug]: change /src/index.js to .jsx and remove Strict Mode
### Describe the Bug

I would like to propose we change /src/index.js to /src/index.jsx by doing this:

```jsx
import React from "react";
import ReactDOM from "react-dom/client";
import App from "./App";
import reportWebVitals from "./reportWebVitals";
// import routes from "./routes";
import * as serviceWorker from "./serviceWorker";
// import 'bootstrap/dist/css/bootstrap.css';

ReactDOM.createRoot(document.getElementById("root")).render(<App />);
// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();

serviceWorker.unregister();
``` 

instead of 

```js
import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';

const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```

### Steps to Reproduce

N/A

### Screenshots

_No response_

### Which device are you using?

Linux VM

### Which operating system are you using?

Linux x64, Debian 11 Base

### Which browser are you using?

Firefox 115

### Additional Context

This will also remove Strict Mode from the render methods.

### Contribute

- [ ] I am willing to contribute and submit a pull request

### PR DESCRIPTION
**Pull request by Piyush-Deshmukh**
Closes #287 

**_Renamed "index.js" to "index.jsx" and removed ReactStrictMode._**

### CODE REVIEW

**Comment by gbowne1** [comment]
@Piyush-Deshmukh Did you test that the app still renders and no errors?

**Comment by gbowne1** [comment]
@Piyush-Deshmukh Also make sure that the App.js has index.jsx in the imports as well.  If not you will need to rename the index.js to index.js in the imports of App.js and commit that to the branch you made your PR on.

**Comment by Piyush-Deshmukh** [comment]
**TL;DR :** I don't think there are any errors after making changes.

@gbowne1 Here are the things I experienced after making changes in the code :

1. I had to restart the development server due the changes made in "index" file.
2. Then I found errors which were previously there and are "out of scope of this issue or pull request".image

3. > Also make sure that the App.js has index.jsx in the imports as well. If not you will need to rename the index.js to index.js in the imports of App.js and commit that to the branch you made your PR on.

     There are no imports in client/src/App.js for index.jsx file as "index.jsx" file is the entry point (or the parent file).


<br></br>
**_Other Screenshots :_**

image


image


image

**Comment by gbowne1** [comment]
@Piyush-Deshmukh 

Could you make an issue for the localStorage utility issue and show those screenshots there?  Those are out of scope for this particular issue.

Oh, that's right, not in App.js my bad.. well there should be somewhere in the code that index.js import exists.

The prettier issue is a minor one.  just needs prettier writing the fixes to the project.  Also out of scopefor this issue..  There is a issue for re-linting and using prettier on the project and is more of a todo issue.

**Comment by Piyush-Deshmukh** [comment]
> Could you make an issue for the localStorage utility issue and show those screenshots there? Those are out of scope for this particular issue.

   **_What should be the title and other details of the issue other than screenshots ?_**

<hr>

~ Also if the review of this pull request is finished could you please merge this pr ?

**Comment by gbowne1** [comment]
@Piyush-Deshmukh 

Just so you know, our PR's require 3 approvals for merge as I am not the only maintainer. All of our PR's require approvals from @pawel975, myself and @manuel12 before merging.  5 successful merges is the minimum number required to be added as a collaborator. 10 for maintainer.

I will be doing a review of this PR. momentarily.

**Comment by gbowne1** [comment]
you can title the new issue as being issue with localstorage utility, and describing the issues you are seeing in the browser console.

**Review by gbowne1 — APPROVED** [review]
Checked out this change locally for testing.

It does not affect the app significantly, and as a note here we do not need React Strict Mode for our app to work properly.

Approving this PR for merge.

**Comment by gbowne1** [comment]
@balijik I'm not sure about actually merging this PR, what do you think?

**Comment by manuel12** [comment]
This seems to work when cloning and testing balajik's repo. Thanks! Will merge so we can get this out of the way :)

**Comment by gbowne1** [comment]
@manuel12 @balajik  with today's merge, I think we are now out of React.StrictMode :-| might wanna go back and dig this out of the commit.. the other stuff in this PR, might be ok.  Moar testing needed.
